src/main/java/io/cdap/plugin/gcp/bigquery/source/CdapBigQueryInputFormat.java

```diff
@@ -0,0 +1,175 @@
+package io.cdap.plugin.gcp.bigquery.source;
+
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableReference;
+import com.google.cloud.hadoop.io.bigquery.AbstractBigQueryInputFormat;
+import com.google.cloud.hadoop.io.bigquery.AvroRecordReader;
+import com.google.cloud.hadoop.io.bigquery.BigQueryConfiguration;
+import com.google.cloud.hadoop.io.bigquery.BigQueryHelper;
+import com.google.cloud.hadoop.io.bigquery.BigQueryStrings;
+import com.google.cloud.hadoop.io.bigquery.Export;
+import com.google.cloud.hadoop.io.bigquery.ExportFileFormat;
+import com.google.cloud.hadoop.io.bigquery.NoopFederatedExportToCloudStorage;
+import com.google.cloud.hadoop.io.bigquery.ShardedExportToCloudStorage;
+import com.google.cloud.hadoop.io.bigquery.UnshardedExportToCloudStorage;
+import com.google.cloud.hadoop.util.ConfigurationUtil;
+import com.google.cloud.hadoop.util.HadoopToStringUtil;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.apache.avro.generic.GenericData;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * BigQuery input format, splits query from the configuration into list of queries
+ * in order to create input splits.
+ */
+public class CdapBigQueryInputFormat extends AbstractBigQueryInputFormat<LongWritable, GenericData.Record> {
+  private static final Logger LOG = LoggerFactory.getLogger(CdapBigQueryInputFormat.class);
+
+  private InputFormat<LongWritable, Text> delegateInputFormat;
+
+  @Override
+  public ExportFileFormat getExportFileFormat() {
+    return ExportFileFormat.AVRO;
+  }
+
+  @Override
+  public List<InputSplit> getSplits(JobContext context) throws IOException, InterruptedException {
+    final Configuration configuration = context.getConfiguration();
+    BigQueryHelper bigQueryHelper;
+    try {
+      bigQueryHelper = getBigQueryHelper(configuration);
+    } catch (GeneralSecurityException gse) {
+      LOG.error("Failed to create BigQuery client", gse);
```

Comment:
Avoid log and rethrow
https://wiki.cask.co/display/CE/Exceptions

Reply:
Done.